extension/ackextension/inmemory.go

```diff
@@ -0,0 +1,108 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ackextension // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/ackextension"
+
+import (
+	"context"
+	"sync/atomic"
+
+	lru "github.com/hashicorp/golang-lru/v2"
+	"go.opentelemetry.io/collector/component"
+)
+
+// InMemoryAckExtension is the in-memory implementation of the AckExtension
+// When MaxNumPartition is reached, the acks associated with the least recently used partition are evicted.
+// When MaxNumPendingAcksPerPartition is reached, the least recently used ack is evicted
+type InMemoryAckExtension struct {
+	partitionMap                  *lru.Cache[string, *ackPartition]
+	maxNumPendingAcksPerPartition uint64
+}
+
+func newInMemoryAckExtension(conf *Config) *InMemoryAckExtension {
```

Comment:
Why is this "in memory" if we pass config with StorageID? I think it should be just `ackExtension` that currently ignores `storage` config (that should be documented), not a different type of extension. Otherwise, it's confusing

Reply:
I intended the code right now to be temporary and assign the inMemory implementation of ackExtension regardless of what the user chooses as the storage type, but I can see where this is confusing. I added the following 

`func createExtension(_ context.Context, _ extension.CreateSettings, cfg component.Config) (extension.Extension, error) {
	if cfg.(*Config).StorageID == nil {
		return newInMemoryAckExtension(cfg.(*Config)), nil
	}
	
	return nil, nil
}
`

to make it more clear. Let me know what you think.